app/account/views.py

```diff
@@ -189,6 +191,35 @@ def confirm(token):
     return redirect(url_for('main.index'))
 
 
+@account.route('/join-from-invite/<int:user_id>/<token>',
+               methods=['GET', 'POST'])
+def join_from_invite(user_id, token):
+    """
+    Confirm new user's account with provided token and prompt them to set
+    a password.
+    """
+    if current_user is not None and current_user.is_authenticated():
+        flash('You have already joined.', 'error')
+        return redirect(url_for('main.index'))
+
+    new_user = User.query.filter_by(id=user_id).first()
+    if new_user is None:
+        return redirect(404)
+
+    if new_user.confirm_account(token):
+            form = CreatePasswordForm()
+            if form.validate_on_submit():
+                new_user.password = form.password.data
+                db.session.add(new_user)
+                db.session.commit()
+                flash('Your password has been set.', 'success')
+                return redirect(url_for('account.manage'))
```

Comment:
Let's redirect them to an index page? Not sure they need to go to their account management settings right after landing in the app—probably a less optimal onboarding experience.